src/main/rx/ghst.c

```diff
@@ -0,0 +1,312 @@
+#define GHST_FRAME_LENGTH_FRAMELENGTH   1
+#define GHST_FRAME_LENGTH_TYPE_CRC      1
+
+// called from telemetry/ghst.c
+void ghstRxWriteTelemetryData(const void *data, int len)
+{
+    len = MIN(len, (int)sizeof(telemetryBuf));
+    memcpy(telemetryBuf, data, len);
+    telemetryBufLen = len;
+}
+
+void ghstRxSendTelemetryData(void)
+{   
+    // if there is telemetry data to write
+    if (telemetryBufLen > 0) {
+        serialWriteBuf(serialPort, telemetryBuf, telemetryBufLen);
+        telemetryBufLen = 0; // reset telemetry buffer
+    }
+}
+
+STATIC_UNIT_TESTED uint8_t ghstFrameCRC(ghstFrame_t *pGhstFrame)
+{
+    // CRC includes type and payload
+    uint8_t crc = crc8_dvb_s2(0, pGhstFrame->frame.type);
+    for (int i = 0; i < pGhstFrame->frame.len - GHST_FRAME_LENGTH_TYPE_CRC - 1; ++i) {
+        crc = crc8_dvb_s2(crc, pGhstFrame->frame.payload[i]);
+    }
+    return crc;
+}
+
+// Receive ISR callback, called back from serial port
+STATIC_UNIT_TESTED void ghstDataReceive(uint16_t c, void *data)
+{
+    UNUSED(data);
+
+    static uint8_t ghstFrameIdx = 0;
+    const timeUs_t currentTimeUs = microsISR();
+
+    if (cmpTimeUs(currentTimeUs, ghstRxFrameStartAtUs) > GHST_MAX_FRAME_TIME_US) {
+        // Character received after the max. frame time, assume that this is a new frame
+        ghstFrameIdx = 0;
+    }
+
+    if (ghstFrameIdx == 0) {
+        // timestamp the start of the frame, to allow us to detect frame sync issues
+        ghstRxFrameStartAtUs = currentTimeUs;
+    }
+
+    // assume frame is 5 bytes long until we have received the frame length
+    // full frame length includes the length of the address and framelength fields
+    const int fullFrameLength = ghstFrameIdx < 3 ? 5 : ghstIncomingFrame.frame.len + GHST_FRAME_LENGTH_ADDRESS + GHST_FRAME_LENGTH_FRAMELENGTH;
+
+    if (ghstFrameIdx < fullFrameLength) {
+        ghstIncomingFrame.bytes[ghstFrameIdx++] = (uint8_t)c;
+        if (ghstFrameIdx >= fullFrameLength) {
+            ghstFrameIdx = 0;
+
+            // NOTE: this data is not yet CRC checked, nor do we know whether we are the correct recipient, this is
+            // handled in ghstFrameStatus
+            memcpy(&ghstValidatedFrame, &ghstIncomingFrame, sizeof(ghstIncomingFrame));
+            ghstFrameAvailable = true;
+
+            // remember what time the incoming (Rx) packet ended, so that we can ensure a quite bus before sending telemetry
+            ghstRxFrameEndAtUs = microsISR();
+        }
+    }
+}
+
+static bool shouldSendTelemetryFrame(void)
+{
+    const timeUs_t now = micros();
+    const timeUs_t timeSinceRxFrameEndUs = cmpTimeUs(now, ghstRxFrameEndAtUs);
+    return telemetryBufLen > 0 && timeSinceRxFrameEndUs > GHST_RX_TO_TELEMETRY_MIN_US && timeSinceRxFrameEndUs < GHST_RX_TO_TELEMETRY_MAX_US;
+}
+
+static void ghstIdle(void)
+{
+    if (ghstTransmittingTelemetry) {
+        ghstTransmittingTelemetry = false;
+    }
+}
+
+uint8_t ghstFrameStatus(rxRuntimeConfig_t *rxRuntimeState)
```

Comment:
So, we don't ever report `RX_FRAME_FAILSAFE`. I guess that's by design by IRC, but shouldn't we apply some logic here based on frame drop rate?